packages/core/src/codewhispererChat/controllers/chat/controller.ts

```diff
@@ -656,12 +655,24 @@ export class ChatController {
                 request
             )}`
         )
-        let response: GenerateAssistantResponseCommandOutput | undefined = undefined
+        let response: MessengerResponseType | undefined = undefined
         session.createNewTokenSource()
         try {
             this.messenger.sendInitalStream(tabID, triggerID)
             this.telemetryHelper.setConversationStreamStartTime(tabID)
-            response = await session.chat(request)
+            if (isSsoConnection(AuthUtil.instance.conn)) {
+                const { $metadata, generateAssistantResponseResponse } = await session.chatSso(request)
+                response = {
+                    $metadata: $metadata,
+                    message: generateAssistantResponseResponse,
+                }
+            } else {
```

Comment:
If we aren't connected (AuthUtil.instance.conn is undefined), we will take the IAM path. Will this provide any confusing errors or log messages?

Reply:
we never get to this point if the Auth is not valid. 

In the same function call, this code block guards against this

```
const credentialsState = await AuthUtil.instance.getChatAuthState()

if (
   !(credentialsState.codewhispererChat === 'connected' && credentialsState.codewhispererCore === 'connected')
) {
    await this.messenger.sendAuthNeededExceptionMessage(credentialsState, tabID, triggerID)
    return
}
 ```